src/test-suite/sanity_check.sh.cmake

```diff
@@ -48,10 +48,148 @@ then
     d=$(diff <(sed -e 's/^ *#[0-9]* *//g' ${ORIG_FILE}) <(sed -e 's/^ *#[0-9]* *//g' ${TEMP}/${VIR_FILE}))
     if [ "$d" == "" ]
     then
-        echo -e "${GREEN}${BOLD}[Test Passed] Vectors of  Oracle and Current version are Identical.${NC}"
+        echo -e "${GREEN}${BOLD}[Test Passed] Vectors of  Oracle and Current version of p-level are Identical.${NC}"
+    else
+        echo -e "$(tput bold)${RED}[Test Failed] Vectors of  Oracle and Current version of p-level are Different.${NC}"
+        exit 1
+    fi
+else
+    echo -e "$(tput bold)${RED}[Error] No embeddings are generated.${NC}"
+    exit 1
+fi
+
+
+ORIG_FILE_F=oracle/${EncodingType}_${SEED_VERSION}_f/ir2vec.txt
+VIR_FILE_F=ir2vec_f.txt
+while IFS= read -r d
+do
+../bin/ir2vec -${PASS} -vocab $Absolute_path_of_RepresentationFile -level f -o ${VIR_FILE_F} ${d} &> /dev/null
+done < index-${SEED_VERSION}.files
+
+TEMP_F=temp_${EncodingType}_${SEED_VERSION}_f
+
+if ls *${VIR_FILE_F} 1> /dev/null 2>&1
+then
+    mkdir -p ${TEMP_F}
+    mv *${VIR_FILE_F} ${TEMP_F}/
+
+    d=$(diff <(sed -e 's/^ *#[0-9]* *//g' ${ORIG_FILE}) <(sed -e 's/^ *#[0-9]* *//g' ${TEMP}/${VIR_FILE}))
+    if [ "$d" == "" ]
+    then
+        echo -e "${GREEN}${BOLD}[Test Passed] Vectors of  Oracle and Current version of f-level are Identical.${NC}"
+
+    else
+        echo -e "$(tput bold)${RED}[Test Failed] Vectors of  Oracle and Current version of f-level are Different.${NC}"
+        exit 1
+    fi
+else
+    echo -e "$(tput bold)${RED}[Error] No embeddings are generated.${NC}"
+    exit 1
+fi
+
+
+ORIG_FILE_ONDEMAND=oracle/${EncodingType}_${SEED_VERSION}_onDemand/ir2vec.txt
+VIR_FILE_ONDEMAND=ir2vec_onDemand.txt
+while IFS= read -r d
+do
+../bin/ir2vec -${PASS} -vocab $Absolute_path_of_RepresentationFile -level f -funcName=main -o ${VIR_FILE_ONDEMAND} ${d} &> /dev/null
```

Comment:
Can you move this to a loop that iterates over an array of different function names?